sequencer/src/bin/espresso-dev-node.rs

```diff
@@ -148,52 +162,130 @@ async fn main() -> anyhow::Result<()> {
         SEQUENCER_VERSION,
     ));
 
-    // Run the prover service. These code are basically from `hotshot-state-prover`. The difference
-    // is that here we don't need to fetch the `stake table` from other entities.
-    // TODO: Remove the redundant code.
-    let proving_key =
-        spawn_blocking(move || Arc::new(load_proving_key(STAKE_TABLE_CAPACITY_FOR_TEST as usize)))
-            .await;
-    let relay_server_client =
-        Client::<ServerError, SequencerVersion>::new(relay_server_url.clone());
-
-    let provider = Provider::<Http>::try_from(url.to_string()).unwrap();
+    let provider = Provider::<Http>::try_from(url.as_str()).unwrap();
     let chain_id = provider.get_chainid().await.unwrap().as_u64();
 
+    let wallet = MnemonicBuilder::<English>::default()
+        .phrase(cli_params.mnemonic.as_str())
+        .index(cli_params.account_index)
+        .expect("error building wallet")
+        .build()
+        .expect("error opening wallet")
+        .with_chain_id(chain_id);
+
     let update_interval = Duration::from_secs(20);
     let retry_interval = Duration::from_secs(2);
+    let prover_port = cli_params
+        .prover_port
+        .unwrap_or(pick_unused_port().unwrap());
```

Comment:
```suggestion
        .unwrap_or_else(|| pick_unused_port().unwrap());
```

Reply:
Any reason why we prefer `.unwrap_or_else`?

Reply:
`unwrap_or()` evaluates even if the Option is Some. While, `unwrap_or_else()` only evaluates when the Option is None. unwrap_or() performs eager evaluation, while unwrap_or_else() does lazy evaluation.
So, in this case unwrap_or() would pick a port even if `Option` is `Some`

Reply:
Thanks. I didn't know that

Reply:
e92c71737d74e7e3db24e017de9bf791f5bfaaac